test/no-hot/no-hot.test.js

```diff
@@ -0,0 +1,29 @@
+'use strict';
+const { run } = require('../utils/test-utils');
+const { stat } = require('fs');
+const { resolve } = require('path');
+describe('no-hot flag', () => {
+    it('should be successful when --no-hot is passed', (done) => {
+        const { stderr, stdout } = run(__dirname, ['--no-hot']);
+        expect(stderr).toBeFalsy();
+        expect(stdout).toBeTruthy();
+        stat(resolve(__dirname, './bin/main.js'), (err, stats) => {
+            expect(err).toBe(null);
+            expect(stats.isFile()).toBe(true);
+            done();
+        });
+    });
+
+    it('should warn when --hot and --no-hot both are passed', (done) => {
+        const { stderr, stdout } = run(__dirname, ['--no-hot', '--hot']);
+        expect(stderr).toContain(
+            '[webpack-cli] You provided both --hot and --no-hot. We will use only the last of these flags that you provided in your CLI arguments',
+        );
+        expect(stdout).toBeTruthy();
+        stat(resolve(__dirname, './bin/main.js'), (err, stats) => {
+            expect(err).toBe(null);
+            expect(stats.isFile()).toBe(true);
+            done();
+        });
+    });
+});
```

Comment:
We should test plugins array here (Do we have `HotModuleReplacemnt` here or not), it is very bad tests where we check `stdout` and `stderr` on `toBeFalsy` and `toBeTruthy`

Reply:
like we check here 
https://github.com/webpack/webpack-cli/blob/e64cf82d227875402ffe3e07b2f1dac845ae1f0c/packages/webpack-cli/__tests__/AdvancedGroup.test.js#L12

Reply:
Yes

Reply:
@evilebottnawi how to extract webpack options from here?

Reply:
@snitin315 I think no way to do this, maybe we can set here `stats: verbose` and check what hot runtime is included?